lib/IRGen/GenDecl.cpp

```diff
@@ -1850,10 +1850,12 @@ LinkInfo LinkInfo::get(const UniversalLinkageInfo &linkInfo,
                        ModuleDecl *swiftModule, const LinkEntity &entity,
                        ForDefinition_t isDefinition) {
   LinkInfo result;
+  ForDefinition_t ForDefinition =
+      ForDefinition_t(swiftModule->isStdlibModule() || isDefinition);
```

Comment:
...please don't shadow `ForDefinition` here, it is exceptionally confusing.

Please check `isDefinition` first, and add a comment explaining what this is about.  I'm not sure I'm totally following how changing the SILLinkage of the entity gets you the right `dllimport`-ness.

Reply:
Will do that.  So, its a pretty heavy hammer: basically, what I'm saying is that any declaration when building the standard library is for the definition.  This prevents it from being marked as an external symbol, which results in the `getIRLinkage` later not marking it as being DLL Import, which means that the DLL storage is now correct.  This happens to work because any declaration which is imported is either marked as such from the ClangImporter (it is from the shims) or is  `@_silgen_name`'ed , and will be marked for import.

Reply:
Okay.  I can live with that hack; please include that as a comment, though.  We should probably integrate these linkage calculations together in the long term.